sklearn/utils/_cython_blas.pyx

```diff
@@ -0,0 +1,199 @@
+# cython: boundscheck=False, wraparound=False, cdivision=True
+
+from cython cimport floating
+
+from scipy.linalg.cython_blas cimport sdot, ddot
+from scipy.linalg.cython_blas cimport sasum, dasum
+from scipy.linalg.cython_blas cimport saxpy, daxpy
+from scipy.linalg.cython_blas cimport snrm2, dnrm2
+from scipy.linalg.cython_blas cimport scopy, dcopy
+from scipy.linalg.cython_blas cimport sscal, dscal
+from scipy.linalg.cython_blas cimport sgemv, dgemv
+from scipy.linalg.cython_blas cimport sger, dger
+from scipy.linalg.cython_blas cimport sgemm, dgemm
+
+
+################
+# BLAS Level 1 #
+################
+
+cdef floating _xdot(int n, floating *x, int incx,
+                    floating *y, int incy) nogil:
```

Comment:
I am wondering about the naming convention, more specifically about the "x" prefix to mark the placeholder for "s" or "d" depending on the type.

I think it would be more readable to just drop the "x" in favor of `_dot` for instance.

WDYT?

Reply:
I stole the name from daal :D
I guess it's to emphasize its fused nature. But I agree it's a bit cumbersome.